qiskit_dynamics/solvers/fixed_step_solvers.py

```diff
@@ -298,18 +300,134 @@ def jax_expm_parallel_solver(
     t_span: Array,
     y0: Array,
     max_dt: float,
+    magnus_order: Optional[int] = 1,
     t_eval: Optional[Union[Tuple, List, Array]] = None,
 ):
-    """Parallel version of jax_expm_solver implemented with JAX parallel operations."""
+    """Parallel version of jax_expm_solver implemented with JAX parallel operations.
+    
+        Args:
+        generator: Generator for the LMDE.
+        t_span: Interval to solve over.
+        y0: Initial state.
+        max_dt: Maximum step size.
+        magnus_order: The expansion order in the Magnus method. Only orders 1, 2 and 3
+        are supported.
+        t_eval: Optional list of time points at which to return the solution.
 
-    def take_step(generator, t, h):
-        eval_time = t + 0.5 * h
-        return jexpm(generator(eval_time) * h)
+    Returns:
+        OdeResult: Results object.
+
+    """
+
+    take_step = get_exponential_take_step(magnus_order, expm_func=jexpm)
 
     return fixed_step_lmde_solver_parallel_template_jax(
         take_step, generator=generator, t_span=t_span, y0=y0, max_dt=max_dt, t_eval=t_eval
     )
 
+def matrix_commutator(m1,m2):
+    """ This function computes the commutator of two matrices.
+    
+    Args: 
+        m1, m2: Two equal-dimension square matrices.
+
+    Returns:
+        commutator: matrix commutator of m1 and m2
+    """
+
+    commutator = m1 @ m2-m2 @ m1
+    
+    return commutator
+
+def get_exponential_take_step(magnus_order, expm_func):
+    
+    """ This function implements the infinitessimal magnus solver at second, 
+    fourth and the sixth order, specified by the user. 
+    See also the documentation of :meth:`scipy_expm_solver` for details.
+
+    Args:
+        magnus_order: The expansion order in the Magnus method. 
+        expm_func: Method of matrix exponentian.
+
+    Returns:
+        take_step: infinitessimal exponential Magnus solver 
+"""
+
+    # if clause based on magnus order
+    if magnus_order==1:
+
+        def take_step(generator, t0, y, h):
+            eval_time = t0 + (h / 2)
+
+            return expm_func(generator(eval_time) * h) @ y
+
+    elif magnus_order==2:
+
+        # second-order step size constants
+        c1=1/2-np.sqrt(3)/6
```

Comment:
This is obviously very minor but change all `1/2` to `0.5`. Obviously a trivial operation but it still saves the construction of the ints and then the division.